common/web/keyboard-processor/src/text/keyboardProcessor.ts

```diff
@@ -217,12 +217,26 @@ export default class KeyboardProcessor extends EventEmitter<EventMap> {
   processKeystroke(keyEvent: KeyEvent, outputTarget: OutputTarget): RuleBehavior {
     var matchBehavior: RuleBehavior;
 
+    // Before keyboard rules apply, check if the left-context is empty.
+    const nothingDeletable = outputTarget.getTextBeforeCaret().kmwLength() == 0 && outputTarget.isSelectionEmpty();
+
     // Pass this key code and state to the keyboard program
     if(this.activeKeyboard && keyEvent.Lcode != 0) {
       matchBehavior = this.keyboardInterface.processKeystroke(outputTarget, keyEvent);
     }
 
-    if(!matchBehavior || matchBehavior.triggerKeyDefault) {
+    // Final conditional component - if someone actually makes a keyboard rule that blocks output
+    // of K_BKSP with an empty left-context or does other really weird things... it's on them.
+    //
+    // We don't expect such rules to appear, but trying to override them would likely result in odd
+    // behavior in cases where such rules actually would appear.  (Though, _that_ should be caught
+    // in the keyboard-review process and heavily discouraged, so... yeah.)
+    if(nothingDeletable && keyEvent.Lcode == Codes.keyCodes.K_BKSP && matchBehavior.triggerKeyDefault) {
+      matchBehavior = this.defaultRuleBehavior(keyEvent, outputTarget, false);
+      matchBehavior.triggerKeyDefault = true;
+      // Force a single `deleteLeft`.
+      matchBehavior.transcription.transform.deleteLeft = 1;
```

Comment:
This value is passed to host apps from `app/webview` as the number of characters to be deleted when the rule behavior is finalized:

https://github.com/keymanapp/keyman/blob/e586dacef47cfd3e21c999eef7798500ca2382d9/web/src/engine/main/src/keymanEngine.ts#L80-L82

... with the final port-of-call occurring here:

https://github.com/keymanapp/keyman/blob/e586dacef47cfd3e21c999eef7798500ca2382d9/web/src/app/webview/src/configuration.ts#L41-L43

The latter block is also utilized by predictive-text when applying and reverting suggestions and has served us well since KMW 12, if I recall correctly.